// Copyright 2018 The Swarming Authors. All rights reserved.
// Use of this source code is governed by the Apache v2.0 license that can be
// found in the LICENSE file.

syntax = "proto3";

package buildbucket.v2;

// Status of a build or a step.
enum Status {
  // Unspecified state. Meaning depends on the context.
  STATUS_UNSPECIFIED = 0;
  // Build was scheduled, but did not start or end yet.
  SCHEDULED = 1;
  // Build/step has started.
  STARTED = 2;
  // A bit present in all completed statuses. A build cannot have this value.
  // Exists for backward compatibility.
  COMPLETED_STATUS_MASK = 4;
  // A build/step ended successfully.
  SUCCESS = 12;  // 8 | COMPLETED_STATUS_MASK
  // A build/step ended unsuccessfully due to its Build.Input,
  // e.g. tests failed, and NOT due to a build infrastructure failure.
  FAILURE = 20;  // 16 | COMPLETED_STATUS_MASK
  // A build/step ended unsuccessfully due to a failure independent of the input,
  // e.g. swarming failed, or the recipe was unable to read the patch from gerrit..
  INFRA_FAILURE = 36;  // 32 | COMPLETED_STATUS_MASK
  // A build was cancelled explicitly, e.g. via an RPC.
  CANCELED = 68;  // 64 | COMPLETED_STATUS_MASK
}

// A Gerrit patchset.
message GerritChange {
  // Gerrit hostname, e.g. "chromium-review.googlesource.com".
  string host = 1;
  // Gerrit project, e.g. "chromium/src".
  string project = 2;
  // Change number, e.g. 12345.
  int64 change = 3;
  // Patch set number, e.g. 1.
  int64 patchset = 4;
}

// A landed Git commit hosted on Gitiles.
message GitilesCommit {
  // Gitiles hostname, e.g. "chromium.googlesource.com".
  string host = 1;
  // Repository name on the host, e.g. "chromium/src".
  string project = 2;
  // Commit HEX SHA1.
  string id = 3;
  // Optional, associated git ref, e.g. "refs/heads/master".
  // NOT a branch name: if specified, must start with "refs/".
  string ref = 4;
}

// A key-value pair of strings.
message StringPair {
  string key = 1;
  string value = 2;
}
